src/deluge/modulation/arpeggiator.cpp

```diff
@@ -647,57 +662,62 @@ void ArpeggiatorBase::render(ArpeggiatorSettings* settings, int32_t numSamples,
 		return;
 	}
 
-	uint32_t gateThresholdSmall = gateThreshold >> 8;
-
-	// Update Sequence Length
-	maxSequenceLength = sequenceLength;
+	// Update live Sequence Length value with the most up to date value from automation
+	maxSequenceLength = (((int64_t)sequenceLength) * kMaxMenuValue + 2147483648) >> 32; // in the range 0-50
 
-	// Update ratchetProbability with the most up to date value from automation
+	// Update live ratchetProbability value with the most up to date value from automation
 	ratchetProbability = ratchProb >> 16; // just 16 bits is enough resolution for probability
 
+	// Update live ratchetAmount value with the most up to date value from automation
 	// Convert ratchAmount to either 0, 1, 2 or 3 (equivalent to a number of ratchets: OFF, 2, 4, 8)
 	uint16_t amount = ratchAmount >> 16;
-	if (amount > 45874) {
+	if (amount > 45874) { // > 35 -> 50
 		ratchetAmount = 3;
 	}
-	else if (amount > 26214) {
+	else if (amount > 26214) { // > 20 -> 34
 		ratchetAmount = 2;
 	}
-	else if (amount > 6553) {
+	else if (amount > 6553) { // > 5 -> 19
 		ratchetAmount = 1;
 	}
-	else {
+	else { // > 0 -> 4
 		ratchetAmount = 0;
 	}
 
+	uint32_t gateThresholdSmall = gateThreshold >> 8;
+
 	if (isRatcheting) {
 		// shorten gate in case we are ratcheting (with the calculated number of ratchet notes)
 		gateThresholdSmall = gateThresholdSmall >> ratchetNotesMultiplier;
 	}
 
+	uint32_t maxGate = 16777216;
```

Comment:
```suggestion
	uint32_t maxGate = 1 << 25;
```

make sure I got the shift amount right, but i think that's clearer.

Reply:
It is 1<< 24. If you can edit your proposal so i can confirm it

Reply:
Fixed